core/src/software/aws/toolkits/core/ToolkitClientManager.kt

```diff
@@ -121,18 +149,48 @@ abstract class ToolkitClientManager {
         @Suppress("UNCHECKED_CAST")
         return builder
             .httpClient(sdkHttpClient())
-            .credentialsProvider(credProvider)
             .region(region)
-            .overrideConfiguration { configuration ->
-                configuration.putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, userAgent)
-                configuration.retryPolicy(RetryMode.STANDARD)
-            }
             .apply {
+                val clientOverrideConfig = ClientOverrideConfiguration.builder()
+
+                if (credProvider != null) {
+                    credentialsProvider(credProvider)
+                }
+
+                if (tokenProvider != null) {
+                    val tokenMethod = builderMethod.returnType.methods.find {
+                        it.name == "tokenProvider" &&
+                            it.parameterCount == 1 &&
+                            it.parameters[0].type.name == "software.amazon.awssdk.auth.token.credentials.SdkTokenProvider"
+                    }
+
+                    if (tokenMethod == null) {
+                        LOG.warn { "Ignoring bearer provider parameter for ${sdkClass.qualifiedName} since it's not a supported client attribute" }
+                    } else {
+                        tokenMethod.invoke(this, tokenProvider)
+                        clientOverrideConfig.nullDefaultProfileFile()
```

Comment:
How does this one reset once I switch to credentials?

Reply:
like existing clients, a new client is instantiated when credentials are changed